src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
+        Node<T> newNode = new Node<>(value, tail, null);
+        if (size == 0) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Can't find element by this index");
+        }
+        Node<T> newNode = new Node<>(value, null, null);
+        if (index == 0) {
+            if (head == null) {
+                head = newNode;
+                tail = newNode;
+            } else {
+                newNode.next = head;
+                head.prev = newNode;
+                head = newNode;
+            }
+        } else if (index == size) {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> current = findNode(index);
+            newNode.prev = current.prev;
+            newNode.next = current;
+            current.prev.next = newNode;
+            current.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        Node<T> current = findNode(index);
+        T oldItem = current.item;
+        Node<T> newNode = new Node<>(value, current.prev, current.next);
+        if (index == 0) {
+            head = newNode;
+            if (head.next != null) {
+                head.next.prev = head;
+            }
+        } else if (index == size - 1) {
+            tail = newNode;
+            if (tail.prev != null) {
+                tail.prev.next = tail;
+            }
+        } else {
+            current.prev.next = newNode;
+            current.next.prev = newNode;
+        }
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return removeObject(findNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        while (current != null) {
+            if ((object == null && current.item == null)
+                    || (object != null && object.equals(current.item))) {
+                removeObject(current);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void indexCheck(int index) {
+        if (index >= size || index < 0) {
```

Comment:
Private methods and classes should be after public ones in your class. Move the 'Node' class and private methods after the public methods.